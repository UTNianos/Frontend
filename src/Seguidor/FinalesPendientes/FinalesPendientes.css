.PendientesList
{
  border: 1px solid gainsboro;
  width: 90%;
  margin-top: 30px;
  padding-left: 30px;
}