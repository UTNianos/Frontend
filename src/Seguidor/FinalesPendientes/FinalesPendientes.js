import React from 'react';
import { List } from 'antd';
import './FinalesPendientes.css';

const FinalesPendientes = ({materias, updateEstado}) => {

  const pendientesDeFinal = materias.filter(m => m.final === true && m.status == 3);

  return(
  <div className="PendientesList">
  
    <List
    itemLayout="horizontal"
    dataSource={pendientesDeFinal}
    renderItem={item => (
      <List.Item actions={[<a>Aprobada</a>]}>
        <List.Item.Meta
          title={<a href="https://ant.design">{item.name}</a>}/>
      </List.Item>
		)}
	/>  
  </div>
  );

}


export default FinalesPendientes;
