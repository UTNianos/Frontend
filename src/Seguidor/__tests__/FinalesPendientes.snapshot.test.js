import React from 'react';
import renderer from 'react-test-renderer';
import FinalesPendientes from '../FinalesPendientes/FinalesPendientes';


describe('<FinalesPendientes />', () => {

  const updateEstado = jest.fn();

  it('Sin materias.', () => {

    const tree = renderer.create(<FinalesPendientes materias={[]} updateEstado={updateEstado} />).toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('Con materias.', () => {

    const materias = [{ cursada: false, final: true, id: '416', name: 'Materia 1', status: 3, year: '0' }, { cursada: false, final: true, id: '415', name: 'Materia 2', status: 3, year: '0' }];
    const tree = renderer.create(<FinalesPendientes materias={materias} updateEstado={updateEstado} />).toJSON();

    expect(tree).toMatchSnapshot();
  })

})
