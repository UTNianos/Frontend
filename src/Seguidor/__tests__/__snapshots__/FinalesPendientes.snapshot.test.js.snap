// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FinalesPendientes /> Con materias. 1`] = `
<div
  className="PendientesList"
>
  <div
    className="ant-list ant-list-split"
  >
    <div>
      <div
        className="ant-spin-nested-loading"
        style={null}
      >
        <div
          className="ant-spin-container"
        >
          <div
            className="ant-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="http://utnianos.com"
                  >
                    Materia 1
                  </a>
                </h4>
              </div>
            </div>
            <ul
              className="ant-list-item-action"
            >
              <li>
                <a
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Aprobada
                </a>
              </li>
            </ul>
          </div>
          <div
            className="ant-list-item"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  <a
                    href="http://utnianos.com"
                  >
                    Materia 2
                  </a>
                </h4>
              </div>
            </div>
            <ul
              className="ant-list-item-action"
            >
              <li>
                <a
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Aprobada
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<FinalesPendientes /> Sin materias. 1`] = `
<div
  style={
    Object {
      "marginTop": "60px",
      "textAlign": "center",
    }
  }
>
  <h1>
    No existen materias pendientes de final.
  </h1>
  <h3>
    No tenés materias pendientes de final. Una vez que termines la cursada de una materia y la marques como 
    <strong>
      firmada
    </strong>
     la misma aparecera en este área.
  </h3>
</div>
`;
